test/Libraries/RevitNodesTests/Elements/DocumentTests.cs

```diff
@@ -17,5 +19,40 @@ public void Current()
             Assert.IsTrue(doc.IsFamilyDocument);
         }
 
+        [Test]
+        [TestModel(@".\Document\DocumentPurgeUnusedTest.rvt")]
+        public void CanRecursivelyPurgeUnusedElementsFromDocument()
+        {
+            // Arrange
+            int expectedPurgedElementNumber = 396;
+            string expectedPurgeMessageFirstRun = string.Format(Revit.Properties.Resources.PurgedElements, expectedPurgedElementNumber);
+            string expectedPurgeMessageSecondRun = Revit.Properties.Resources.NoElementsToPurge;
+
+            // Act
+            var document = Document.Current;
+            var resultFirstRun = document.PurgeUnused(true);
+            var resultSecondRun = document.PurgeUnused(true);
+
+            // Assert
+            Assert.AreEqual(expectedPurgeMessageFirstRun, resultFirstRun);
+            Assert.AreEqual(expectedPurgeMessageSecondRun, resultSecondRun);
+        }
+
+        [Test]
+        [TestModel(@".\Document\DocumentPurgeUnusedTest.rvt")]
+        public void CanPurgeUnusedElementsFromDocument()
+        {
+            // Arrange
+            int expectedPurgedElementNumber = 393;
```

Comment:
we'll need to make & check a list of ids of elements deleted, i suggest we prep a super-minimal RVT file where <10 elements are purged.

Reply:
it should also include a few elements that would only be purged on a second run, so we check we're not doing it recursively.